### PR TITLE
Fix the issue of restarting DataNode to clean up InvalidDataRegion

### PR DESCRIPTION
**Pull request by 133tosakarin**
**### What changes were proposed in this pull request?**

In the previous implementation, only invalid data in the consensus directory was deleted when restarting, resulting in region data that does not belong to the datanode still existing in the data directory.

This PR will completely clean up invalid regions when restarting.

### CODE REVIEW

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/DataNode.java [hunk 1]
```diff
@@ -553,33 +554,92 @@ private void removeInvalidRegions(List<ConsensusGroupId> dataNodeConsensusGroupI
         SchemaRegionConsensusImpl.getInstance().getAllConsensusGroupIdsWithoutStarting().stream()
             .filter(consensusGroupId -> !dataNodeConsensusGroupIds.contains(consensusGroupId))
             .collect(Collectors.toList());
-    removeInvalidDataRegions(invalidDataRegionConsensusGroupIds);
-    removeInvalidSchemaRegions(invalidSchemaRegionConsensusGroupIds);
+    Map<String, List<DataRegionId>> localDataRegionInfo =
+        StorageEngine.getInstance().getLocalDataRegionInfo();
+    logger.info("localDataRegionInfo: {}", localDataRegionInfo);
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/DataNode.java:559]: remove it

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/DataNode.java [hunk 2]
```diff
@@ -553,33 +554,92 @@ private void removeInvalidRegions(List<ConsensusGroupId> dataNodeConsensusGroupI
...
     logger.info("Remove invalid dataRegion directories... {}", invalidConsensusGroupId);
     for (ConsensusGroupId consensusGroupId : invalidConsensusGroupId) {
       File oldDir =
           new File(
               DataRegionConsensusImpl.getInstance()
                   .getRegionDirFromConsensusGroupId(consensusGroupId));
-      removeRegionsDir(oldDir);
+      List<String> allLocalFilesFolders = TierManager.getInstance().getAllLocalFilesFolders();
+      localDataRegionInfo.forEach(
+          (storageGroup, dataRegionIds) -> {
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/DataNode.java:575]: use database

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/DataNode.java [hunk 3]
```diff
@@ -553,33 +554,92 @@ private void removeInvalidRegions(List<ConsensusGroupId> dataNodeConsensusGroupI
...
     logger.info("Remove invalid schemaRegion directories... {}", invalidConsensusGroupId);
     for (ConsensusGroupId consensusGroupId : invalidConsensusGroupId) {
       File oldDir =
           new File(
               SchemaRegionConsensusImpl.getInstance()
                   .getRegionDirFromConsensusGroupId(consensusGroupId));
-      removeRegionsDir(oldDir);
+      localDataRegionInfo.forEach(
+          (storageGroup, dataRegionIds) -> {
+            for (DataRegionId dataRegionId : dataRegionIds) {
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/DataNode.java:616]: schemaregionid

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/DataNode.java (hunk 1 above)
Comment (133tosakarin) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/DataNode.java:559]: ok

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/DataNode.java (hunk 2 above)
Comment (133tosakarin) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/DataNode.java:575]: ok

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/DataNode.java (hunk 3 above)
Comment (133tosakarin) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/DataNode.java:616]: ok

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**